src/App.js

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from "react";
+import { BrowserRouter as Router , Routes, Route, Link, Navigate } from "react-router-dom";
+import './usercss/user.css'; 
+import './style.css'; 
+
+import CreateUser from "./components/user/create.js";
+import Login from "./components/user/login.js";
+import CreateProduct from "./components/product/create.js";
+import CreateBlog from "./components/blog/create.js"
+import Category from "./components/blog/category.js"
+import BlogList from "./components/blog/list.js"
+import Tags from "./components/blog/tags.js"
+import { isAuthenticated } from './authService';
+
+function App() {
+  const [authenticated, setAuthenticated] = useState(null);
+
+  useEffect(() => {
+    const checkAuthentication = async () => {
+      try {
+        const result = await isAuthenticated();
+        console.log(result);
+        setAuthenticated(result);
+      } catch (error) {
+        console.error("Error checking authentication:", error);
+        setAuthenticated(false); // Set authenticated to false in case of an error
+      }
+    };
+
+    checkAuthentication();
+  }, [authenticated]);
+
+  if (authenticated === null) {
+     
+    return null;
+  }
+
+  return (
+    <>
+      <Router>
+        
+
+       
+              <Routes>
+                
+      
+                <Route path="/user/create" element={<CreateUser />} />
+                <Route path="/login" element={<Login />} />
+                 
+                {authenticated ? (
```

Comment:
instead of repeating this pattern again and again you  can create a component which internally do this and use that component here.